File: backend/app/schemas/batch.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class CreateBatchRequest(BaseModel):
    original_folder_name: str = Field(min_length=1, max_length=255)


class BatchResponse(BaseModel):
    id: UUID
    batch_code: str
    original_folder_name: str
    status: str
    total_images: int
    uploaded_images: int
    processed_images: int
    review_count: int
    ok_count: int
    ng_count: int
    failed_count: int
    created_at: datetime


class UploadedImageResponse(BaseModel):
    id: UUID
    original_filename: str
    status: str
    duplicate: bool = False


class BatchImageResponse(BaseModel):
    image_id: UUID
    original_filename: str
    image_status: str
    width: int
    height: int
    file_size: int
    created_at: datetime
    customer_id_ai: str | None
    meter_reading_ai: str | None
    final_confidence: float | None
    review_status: str | None
    final_customer_id: str | None
    final_meter_reading: str | None

File: frontend/types/batch.ts
export type Batch = {
  id: string;
  batch_code: string;
  original_folder_name: string;
  status: string;
  total_images: number;
  uploaded_images: number;
  processed_images: number;
  review_count: number;
  ok_count: number;
  ng_count: number;
  failed_count: number;
  created_at: string;
};
export type BatchImage = {
  image_id: string;
  original_filename: string;
  image_status: string;
  width: number;
  height: number;
  file_size: number;
  created_at: string;
  customer_id_ai: string | null;
  meter_reading_ai: string | null;
  final_confidence: number | null;
  review_status: string | null;
  final_customer_id: string | null;
  final_meter_reading: string | null;
};
export type UploadProgress = {
  total: number;
  uploaded: number;
  failed: number;
};
